tests/core/test_gvm.py

```diff
@@ -140,15 +140,15 @@ def test_type_check(caplog):
 
 
 def test_keys(caplog):
-    gvm = GlobalVariableManager()
+    gvm = GlobalVariableManager.instance()
     gvm.set_variable('a', 1)
     gvm.set_variable('ab', 2)
     gvm.set_variable('abc', 3)
     gvm.set_variable('def', 4)
 
     key_start_a = gvm.get_all_keys_starting_with('a')
-    assert (len(gvm.get_all_keys()) == 4)
-    assert (len(key_start_a) == 3)
+    assert (len(gvm.get_all_keys()) == 33)
```

Comment:
I guess the change in number arises from the new singleton concept. In my opinion, there needs to be a way to reset a singleton, and this should be done before every test. Otherwise, this test is obsolete and cannot be run independent from other tests.